src/Microsoft.AspNetCore.Blazor/Components/BlazorComponent.cs

```diff
@@ -175,14 +175,21 @@ private void HandleException(Exception ex)
             Console.Error.WriteLine($"[{ex.GetType().FullName}] {ex.Message}\n{ex.StackTrace}");
         }
 
-        void IHandleEvent.HandleEvent(UIEventHandler handler, UIEventArgs args)
+        void IHandleEvent.HandleEvent(EventBinding binding, UIEventArgs args)
         {
-            handler(args);
+            var task = binding.Invoke(args);
 
             // After each event, we synchronously re-render (unless !ShouldRender())
             // This just saves the developer the trouble of putting "StateHasChanged();"
             // at the end of every event callback.
             StateHasChanged();
+            
+            if (task.Status == TaskStatus.RanToCompletion)
+            {
+                return;
+            }
+
+            task.ContinueWith(ContinueAfterLifecycleTask);
```

Comment:
It seemed like a good idea to to do this the same way as the other async things on `BlazorComponent` - which I overlooked before. Let me know if you think is this wrong.

Reply:
Yes, this looks good!